lib/auth.js

```diff
@@ -0,0 +1,248 @@
+  },
+  MAX_REPEATED_CHARS: {
+    test: (password, options) => {
+      const regexp = new RegExp(`(.)\\1{${options.number},}`);
+      return !regexp.test(password);
+    },
+    hint: options => 'The password can not contain a sequence of more than ' +
+      `${options.number} repeat chars`,
+    options: { number: 2 }
+  },
+  MIN_LOWERCASE_CHARS: {
+    test: (password, option) => (
+      stringIncludesChars(password, unicodeCategories.Ll, option.number)
+    ),
+    hint: options => `The password must contain at least ${options.number} ` +
+      'lowercase chars',
+    options: { number: 1 }
+  },
+  MIN_UPPERCASE_CHARS: {
+    test: (password, options) => (
+      stringIncludesChars(password, unicodeCategories.Lu, options.number)
+    ),
+    hint: options => `The password must contain at least ${options.number} ` +
+      'uppercase chars',
+    options: { number: 1 }
+  },
+  MIN_NUMBERS: {
+    test: (password, options) => {
+      const NUMBERS_RANGE = [[49, 57]];
+      return stringIncludesChars(password, NUMBERS_RANGE, options.number);
+    },
+    hint: options => `The password must contain at least ${options.number} ` +
+      'numbers',
+    options: { number: 1 }
+  },
+  MIN_SPECIAL_CHARS: {
+    test: (password, options) => {
+      const SPECIAL_CHARS_RANGE = [[32, 47], [58, 64], [91, 96], [123, 126]];
+      // https://www.owasp.org/index.php/Password_special_characters
+      return stringIncludesChars(password, SPECIAL_CHARS_RANGE, options.number);
+    },
+    hint: options => `The password must contain at least ${options.number} ` +
+      'special chars: !"#$%&\'()*+,-./`:;<=>?@[\\]^_{|}~',
+    options: { number: 1 }
+  }
+};
+
+const loginTests = {
+  MIN_LENGTH: {
+    test: (login, options) => login.length >= options.minLength,
+    hint: options => 'The login must be longer than or equal to ' +
+      `${options.minLength} chars`,
+    options: { minLength: 6 }
+  },
+  MAX_LENGTH: {
+    test: (login, options) => login.length <= options.maxLength,
+    hint: options => 'The login must be less thanor equal to ' +
+      `${options.maxLength} chars`,
+    options: { maxLength: 50 }
+  },
+  IS_EMAIL: {
+    test: login => {
+      const EMAIL_REGEXP = new RegExp(
+        '^[a-z0-9!#$%&\'*+/=?^_`{|}~-]+(?:.[a-z0-9!#$%&\'*+/=?^_`{|}~-]+)*@(' +
+        '?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?$'
+      );
+      const MAX_DOMAIN_LENGTH = 255;
+      const MAX_LOCAL_PART_LENGTH = 64;
+
+      if (login.includes('@')) {
+        const [localPart, domain] = login.split('@');
+        return domain.length <= MAX_DOMAIN_LENGTH &&
+          localPart.length <= MAX_LOCAL_PART_LENGTH &&
+          EMAIL_REGEXP.test(login);
+      }
+    },
+    hint: () => 'Login must be email address'
+  }
+};
+
+const loginPasswordTests = {
+  LOGIN_INCLUDES_PASSWORD: {
+    test: (login, password) => !login.includes(password),
+    hint: () => 'The login can not contain a password'
```

Comment:
I don't think that you mean 'a password' here (meaning it can be any password), it should probably say 'the password' since you are referring to the password that was provided to you by the user.